File: From dev/office-addins/fraud-email-scanner/src/scanner/services/scannerEngine.ts
// src/services/scannerEngine.ts

import type { EmailModel } from "../models/EmailModel";
import { extractFeatures, type Features } from "./featureExtractor";
import { applyRules, type Flag } from "./ruleEngine";
import { calculateRisk, type RiskLevel } from "./riskScorer";

export interface ScanResult {
  risk_score: number;
  risk_level: RiskLevel;
  flags: Flag[];
  feature_summary: Features;
}

export function scanEmail(email: EmailModel): ScanResult {
  const features = extractFeatures(email);
  const { score, flags } = applyRules(features);
  const risk_level = calculateRisk(score);

  return {
    risk_score: score,
    risk_level,
    flags,
    feature_summary: features,
  };
}

File: security-tools/email-scanner-python/scanner/services/scanner_engine.py
# use relative imports now that scanner is a package
from .feature_extractor import extract_features
from .rule_engine import apply_rules
from .risk_scorer import calculate_risk

def scan_email(email):
    features = extract_features(email)
    score, flags = apply_rules(features)
    risk_level = calculate_risk(score)

    return {
        "risk_score": score,
        "risk_level": risk_level,
        "flags": flags,
        "feature_summary": features
    }